content/posts/2022-11-09-iOS16-lock-screen-dynamic-time/index.md

```diff
@@ -0,0 +1,185 @@
+---
+title: iOS16 Lock Screen Series - Part 2 (Dynamic time)
+image: /images/thumbnails/posts/2022-11-05-iOS-16-lock-screen.jpeg
+publishedOn: '2022-11-09T17:48:30.556Z'
+description: 'This is our second part of the iOS 16 lock screen series. Here we will be learning how to work with date-time and learn the functionality of creating a simple clock that renders time dynamically.'
+category: React Native
+tags:
+  - React Native CLI
+  - React Native
+  - Expo
+  - Reanimated
+  - dayjs
+keywords: react native, expo, react native cli, reanimated, UI, UX, dayjs, clock
+author: Saad
+---
+
+import YoutubeVideo from "../../../src/components/shared/YoutubeVideo/YoutubeVideo";
+import Snack from '../../../src/components/shared/Snack/Snack';
+
+[In our first part](https://www.notjust.dev/blog/2022-11-05-iOS16-lock-screen-ui), we started with the iOS 16 lock screen and we managed to build the awesome UI together. Let’s just review what we have built so far
+
+![Preview](./ui-preview.png)
+
+As you can see from the preview we built 3 main components in our screen,
+
+1. Header section → Where we show the lock icon and date-time.
+2. Notification list
+3. Footer → Where we show the flashlight and camera icon.
+
+As of now our date and time inside header are fixed which is really boring! Let’s change it. This is going to be the second part of the series where we would like to create a dynamic time component that will give us an updated date and time. Just like the clock on your wall.
+
+## Install `dayjs` package
+
+First we will do some prep work. In order to deal with date and time there are lots of libraries which help us make our life easier. In this project, we are going to use [dayjs](https://day.js.org/) library. Why dayjs?
+
+1. It has lots of utility functions for us. Formatting, validating, and parsing dates are really simple!
+2. The library size is really small.
+
+To install this library just run,
+
+```jsx
+npm install dayjs
+```
+
+## Render the date-time
+
+We can render the date and time the way we want it very easily by the help of dayjs library
```

Comment:
The structure of this sentence is really confusing. "by the help" doesn't sound right. Maybe "with the help"?